### PR TITLE
feat(web): zendesk ticket service

### PR DESCRIPTION
**Pull request by stjanilofts**
Today the contact form sends a regular email to the island.is support email address. This instead creates a Zendesk ticket and submits it.

First checks if the user with the email exists. If the user does not exist it creates it. Then it submits a ticket with that user as the requester.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/zendesk-service/src/lib/zendesk.module.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+} from './zendesk.service'
+
+export class ZendeskModule {
+  static register(options: ZendeskServiceOptions): DynamicModule {
+    return {
+      module: ZendeskModule,
+      providers: [
+        {
+          provide: ZENDESK_OPTIONS,
+          useFactory: () => options,
```
Comment (stefanhar) [inline libs/zendesk-service/src/lib/zendesk.module.ts:16]: Maybe useValue is better suited bere?

**Review by stefanhar — COMMENTED** [review]
we should update `.github/CODEOWNERS` file for these new libraries

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -17,6 +17,7 @@
 /libs/api/domains/cms/                                                                        @island-is/kosmos-kaos @island-is/aranja
 /libs/application/templates/party-letter/                                                     @island-is/kosmos-kaos
 /libs/clients/rsk/                                                                            @island-is/kosmos-kaos
+/libs/zendesk-service/                                                                        @island-is/kosmos-kaos
```
Comment (stefanhar) [inline .github/CODEOWNERS:20]: Maybe we should add the communications domain to here as well?

File: libs/clients/zendesk/src/lib/zendesk.service.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+export interface ZendeskServiceOptions {
+  email: string
+  token: string
+  subdomain: string
+}
+
+export class ZendeskService {
+  #api: string
```
Comment (jeremybarbet) [inline libs/clients/zendesk/src/lib/zendesk.service.ts:27]: Is the `#` doing something special?

Comment (stjanilofts) [reply libs/clients/zendesk/src/lib/zendesk.service.ts]: not really, I will remove the #'s..

**Review by stefanhar — APPROVED** [review]
LGTM

**Review by jeremybarbet — APPROVED** [review]
LGTM

**Pull request merged**